aws-lambda/src/databricks_cdk/utils.py

```diff
@@ -1,23 +1,38 @@
 import logging
 import os
+from functools import lru_cache
 from typing import Any, Dict, Optional
 
 import boto3
 from databricks.sdk import AccountClient, WorkspaceClient
 from databricks.sdk.core import Config
 from pydantic import BaseModel
 from requests import request
-from requests.auth import HTTPBasicAuth
 from requests.exceptions import HTTPError
 from tenacity import retry, retry_if_exception, retry_if_exception_type, stop_after_attempt, wait_exponential
 
 logger = logging.getLogger(__name__)
 
 
-USER_PARAM = os.environ.get("USER_PARAM", "/databricks/deploy/user")
-PASS_PARAM = os.environ.get("PASS_PARAM", "/databricks/deploy/password")
+class UnsupportedAuthMethodError(Exception):
+    pass
+
+
 ACCOUNT_PARAM = os.environ.get("ACCOUNT_PARAM", "/databricks/account-id")
+CLIENT_SECRET_PARAM = os.environ.get("CLIENT_SECRET_PARAM", "/databricks/deploy/client-secret")
+
+PASS_PARAM = os.environ.get("PASS_PARAM")
+
+# Make sure password based authentication is not used anymore after 10th of July 2024
+if PASS_PARAM is not None:
```

Comment:
Isn't it better to raise an exception if CLIENT_SECRET_PARAM is none? does also catch new cases where people forget to set.